src/status_im/chat/models.cljs

```diff
@@ -59,3 +62,43 @@
   "Just like `update-chat` only implicitely updates timestamp"
   [cofx chat]
   (update-chat cofx (assoc chat :timestamp (:now cofx))))
+
+(defn remove-chat [{:keys [db]} chat-id]
+  (let [{:keys [chat-id group-chat debug?]} (get-in db [:chats chat-id])]
+    (cond-> {:db                      (-> db
+                                          (update :chats dissoc chat-id)
+                                          (update :deleted-chats (fnil conj #{}) chat-id))
+             :delete-pending-messages chat-id}
+      (or group-chat debug?)
+      (assoc :delete-messages chat-id)
+      debug?
+      (assoc :delete-chat chat-id)
+      (not debug?)
+      (assoc :deactivate-chat chat-id))))
+
+(defn group-name-from-contacts [contacts selected-contacts username]
+  (->> (select-keys contacts selected-contacts)
+       vals
+       (map :name)
+       (cons username)
+       (string/join ", ")))
+
+(defn prepare-group-chat
+  [{:keys [current-public-key username]
+    :group/keys [selected-contacts]
+    :contacts/keys [contacts]} group-name]
+  (let [selected-contacts'  (mapv #(hash-map :identity %) selected-contacts)
+        chat-name (if-not (string/blank? group-name)
+                    group-name
+                    (group-name-from-contacts contacts selected-contacts username))
+        {:keys [public private]} (protocol/new-keypair!)]
+    {:chat-id     (random/id)
```

Comment:
We have `random-id` available as coeffect, please use-that instead of side-effect call in the function, otherwise, the function is not pure and deterministic.